### DISCUSSION TITLE
Xbox Console: any way to enable hardware mouse input?

### DISCUSSION THREAD

**Discussion by TheElixZammuto** [MicrosoftEdge/WebView2Feedback#4117] [Q&A]
Hi everyone, I tried succesfully to run a WebView2 on my Xbox console. The only problem is that the cursor does only work when using the gamepad and not an attached hardware mouse. The mouse does work fine on apps like Dev Mode home. Is there any way to make it  work and listen to events? I thought this was possible since the Microsoft Edge does detect the connected mouse and respond to its events.  I'm more than happy to know if there are any workaround to this too (like some native APIs that must be bridged or something similar). Thank you and have a good day!

**Comment by jennifer-nejad** [comment]
Hi @TheElixZammuto! Excited to hear you're trying out WebView2 on Xbox.  Can you tell me more about what your app does and what hardware mouse support would help with?

Unfortunately, Xbox currently doesn't have a way to enable hardware mouse.  There hasn't been a huge ask for it in the past so it hasn't been prioritized, but it can be added to their feature requests if there's a sufficient ask for it.

**Reply by TheElixZammuto** [reply]
Hi @jennifer-jacobs,
I'm the developer of https://github.com/TheElixZammuto/moonlight-xbox, which is an alternative to the Microsoft Wireless Display app, optimized for gaming and low latency. 

I was planning WebView2 support for two features:

1 (which is fairly normal). is to use the WebView2 to make the user able to open the Web UI of the Server that Moonlight uses, called Sunshine: https://github.com/LizardByte/Sunshine

2 (which is very crazy to be honest). Many users (I had multiple features requests on GitHub, Microsft Store reviews, reddit, Twitter and Discord) want to be able to use KB+M connected to the xbox to pass through on their PC, especially on strategy-like games. I was able to make the keyboard work but not the mouse as you already have guessed. Also the app already is capable of emulating KB+M using the gamepad (by moving the controller sticks), but for many users this is not enough.

When I read the news of webview2 available on the Xbox, I was hoping that it was able to read the mouse input like the MS Edge App, so I could instantiate a transparent/hidden WebView2 to read the events and send it to the native app using JavaScript (I know it's very convoluted,  but sometimes you need those types of workarounds)

I understand this is a very very very niche and hacky usecase, but I still think the app feels incomplete without this support and I would love to offer this to my users!

**Reply by champnic** [reply]
I've added this as a feature request: https://github.com/MicrosoftEdge/WebView2Feedback/issues/4207
